src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs

```diff
@@ -31,10 +31,44 @@ internal interface ISyntaxFactsService : ILanguageService
         bool IsPredefinedType(SyntaxToken token, PredefinedType type);
         bool IsPredefinedOperator(SyntaxToken token);
         bool IsPredefinedOperator(SyntaxToken token, PredefinedOperator op);
-        bool IsKeyword(SyntaxToken token);
+
+        /// <summary>
+        /// Returns 'true' if this a 'reserved' keyword for the language.  A 'reserved' keyword is a
+        /// identifier that is always treated as being a special keyword, regardless of where it is
+        /// found in the token stream.  Examples of this are tokens like <see langword="class"/> and
+        /// <see langword="Class"/> in C# and VB respectively.
+        /// 
+        /// Importantly, this does *not* include contextual keywords.  If contextual keywords are
+        /// important for your scenario, use <see cref="IsContextualKeyword"/> or <see
+        /// cref="ISyntaxFactsServiceExtensions.IsReservedOrContextualKeyword"/>.  Also, consider using
+        /// <see cref="ISyntaxFactsServiceExtensions.IsWord"/> if all you need is the ability to know 
+        /// if this is effectively any identifier in the language, regardless of whether the language
+        /// is treating it as a keyword or not.
+        /// </summary>
+        bool IsReservedKeyword(SyntaxToken token);
+
+        /// <summary>
+        /// Returns 'true' if this a 'contextual' keyword for the language.  A 'contextual' keyword
+        /// is a identifier that is only treated as being a special keyword in certain *syntactic*
+        /// contexts.  Examples of this is 'yield' in C#.  This is only a keyword if used as 'yield
+        /// return' or 'yield break'.  Importantly, identifiers like <see langword="var"/>, <see
+        /// langword="dynamic"/> and <see langword="nameof"/> are *not* 'contextual' keywords.  This
+        /// is because they are not treated as keywords depending on the syntactic context around
+        /// them.  Instead, the language always treats them identifiers that have special *semantic*
+        /// meaning if they end up not binding to an existing symbol.
+        /// 
+        /// Or, in other words, the parser must be able to identify these cases in order to be a 
+        /// contextual keyword.  If identification happens afterwards, it's not contextual.
+        /// </summary>
```

Comment:
will update comment.

Reply:
Basically, what does "contextual keyword" mean? Is `await` outside an `async` method a contextual keyword (even though it's an identifier) because the word `await` is/can/is one of the contextual keywords? Or is it not a contextual keyword because it's not in the right context?

Reply:
Thanks!